### PR TITLE
Add covary method to lift SCollection to the specified type

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala [hunk 1]
```diff
@@ -210,6 +210,9 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   // Collection operations
   // =======================================================================
 
+  /** lifts this [[SCollection]] to the specified type */
+  def covary[U]: SCollection[U] = this.asInstanceOf[SCollection[U]]
```
Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: Should it be `private`?

Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: I'm ok with it being `public` it can be useful and it's doesn't seem harmful?

Comment (jto) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: In that case could it have type bounds or an implicit witness ?

Comment (regadas) [reply scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala]: On a second thought yeah this can be harmful.

@jto yeah that sounds good

**Pull request merged**